core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
That might be a bit confusing/error-prone?  I think part of the problem is that NthOrder with n=1 probably behaves unexpectedly for stores that don't require login, so FirstOrder has some stuff to make it behave more correctly for the n=1 case.